streams-contrib/streams-persist-mongo/src/main/java/org/apache/streams/mongo/MongoPersistReader.java

```diff
@@ -236,7 +252,7 @@ public StreamsResultSet readRange(DateTime start, DateTime end) {
 
     @Override
     public boolean isRunning() {
-        return !executor.isTerminated();
+        return !executor.isTerminated() || !executor.isShutdown();
```

Comment:
Should this be && and not ||?  Reading the documentation these states might overlap, depending on how the ExecutorService was shutdown.